components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAO.java

```diff
@@ -90,6 +93,42 @@ public CacheBackedIdPMgtDAO(IdPManagementDAO idPMgtDAO) {
 		return idPMgtDAO.getIdPsSearch(dbConnection, tenantId, tenantDomain,
 				filter);
 	}
+
+    /**
+     * Get all basic identity provider information for a matching filter.
+     *
+     * @param tenantId             tenant Id of the identity provider.
+     * @param expressionConditions filter value list for IdP search.
+     * @param limit                limit per page.
+     * @param offset               offset value.
+     * @param sortOrder            order of IdP ASC/DESC.
+     * @param sortBy               the attribute need to sort.
+     * @return Identity Provider's basic information array.
+     * @throws IdentityProviderManagementServerException Error when getting list of Identity Providers.
+     * @throws IdentityProviderManagementClientException Error when append the filer string.
+     */
+    public List<IdentityProvider> getIdPsSearch(int tenantId, List<ExpressionNode> expressionConditions, int limit,
```

Comment:
IMO this API name is not very meaningful. It should convey that this is returning the basic IdP information that matches the given filter.